addons/fleet/fleet.py

```diff
@@ -1,818 +1,731 @@
+            if model.make_id.name:
+                name = model.make_id.name + ' / ' + name if name else ''
+            result.append((model.id, name))
+        return result
 
+    @api.onchange('make_id')
+    def on_change_make(self):
+        self.image = self.make_id.image
 
-class fleet_vehicle_model_brand(osv.Model):
-    _name = 'fleet.vehicle.model.brand'
-    _description = 'Brand model of the vehicle'
+    name = fields.Char(store=True, required=True)
+    make_id = fields.Many2one('fleet.make', string='Make', oldname='brand_id', required=True, help='Make of the vehicle')
+    vendors = fields.Many2many('res.partner', 'fleet_vehicle_model_vendors', 'model_id', 'partner_id')
+    image = fields.Binary(related='make_id.image', string="Logo", store=True)
+    image_medium = fields.Binary(related='make_id.image_medium', string="Logo (medium)", store=True)
+    image_small = fields.Binary(related='make_id.image_small', string="Logo (small)", store=True)
 
-    _order = 'name asc'
 
-    def _get_image(self, cr, uid, ids, name, args, context=None):
-        result = dict.fromkeys(ids, False)
-        for obj in self.browse(cr, uid, ids, context=context):
-            result[obj.id] = tools.image_get_resized_images(obj.image)
-        return result
+class FleetMake(models.Model):
+    _name = 'fleet.make'
+    _description = 'Make of the vehicle'
+    _order = 'name asc, id asc'
 
-    def _set_image(self, cr, uid, id, name, value, args, context=None):
-        return self.write(cr, uid, [id], {'image': tools.image_resize_image_big(value)}, context=context)
-
-    _columns = {
-        'name': fields.char('Make', required=True),
-        'image': fields.binary("Logo",
-            help="This field holds the image used as logo for the brand, limited to 1024x1024px."),
-        'image_medium': fields.function(_get_image, fnct_inv=_set_image,
-            string="Medium-sized photo", type="binary", multi="_get_image",
-            store = {
-                'fleet.vehicle.model.brand': (lambda self, cr, uid, ids, c={}: ids, ['image'], 10),
-            },
-            help="Medium-sized logo of the brand. It is automatically "\
-                 "resized as a 128x128px image, with aspect ratio preserved. "\
-                 "Use this field in form views or some kanban views."),
-        'image_small': fields.function(_get_image, fnct_inv=_set_image,
-            string="Smal-sized photo", type="binary", multi="_get_image",
-            store = {
-                'fleet.vehicle.model.brand': (lambda self, cr, uid, ids, c={}: ids, ['image'], 10),
-            },
-            help="Small-sized photo of the brand. It is automatically "\
-                 "resized as a 64x64px image, with aspect ratio preserved. "\
-                 "Use this field anywhere a small image is required."),
-    }
+    @api.multi
+    @api.depends('image')
+    def _compute_get_image(self):
+        all_image = tools.image_get_resized_images(self.image)
```

Comment:
@rga-odoo : i think i have to use self.ensure_one() instead of api.one, as compute method return value for image_medium and image_small.Is it proper?